### ISSUE TITLE
Selective download

### ISSUE THREAD

**Issue by JohnVeness** [EugeneLoy/edx-archive#12]
It would be useful if we could limit which pages to download, e.g. by section (as edx-dl allows), or even by subsection or unit.

**Comment by EugeneLoy** [comment]
Hi. I was thinking about adding filtering based on range\pattern\regex. Once this is implemented it should address your request. ETA is next few weeks.

**Comment by EugeneLoy** [comment]
Pages can now be filtered using `--filter-range` option. This is now in master and will be released in `v1.7.0`.
Implementing filters based on page name\section should be thought through more carefully as this information might not be available at current filter-time. Keeping this issue open until this is settled.

**Comment by JohnVeness** [comment]
This sounds good. Is there a way you can show the user what page number will be what, so they can pick which range to include?

**Comment by EugeneLoy** [comment]
Maybe. I was thinking about implementing something like "dry run" that lists what is about to get downloaded without actually downloading it. This might provide relevant info. Anyway, I have to give more thought to it.

**Comment by JohnVeness** [comment]
Many thanks for adding ``--filter-range``, which I have just tried out for the first time.

At the moment, the filter-range option only accepts input like ``from-to``, e.g. ``30-40``. If a single page is required, the user has to use e.g. ``30-30``.

It would be good if it could accept more inputs. E.g. like when printing from Microsoft Word, where these are valid inputs when specifying a page range:

- ``30-40`` means pages 30 to 40 inclusive (as current)
- ``30`` means just page 30
- ``-30`` means all pages up to and including 30 (same as ``1-30``)
- ``30-`` means all pages from 30 until end of list (means the user doesn't need to know how many pages there will be in total)
- ``30,40`` means page 30 and page 40 with nothing in between

Even better if you can accept any other valid range within a comma-separated list, such as ``1,4,7-25,54,70-``.

**Comment by EugeneLoy** [comment]
Yes, I agree, extending this might be a good idea. Will work on this soon.
